chasm/field.go

```diff
@@ -47,21 +49,64 @@ type Field[T any] struct {
 }
 
 type fieldInternal struct {
-	fieldType fieldType
-	value     any // Component | Data | Pointer
+	// These 2 fields are used when node is not created yet.
+	// Don't access them directly outside of this file. Use corresponding getters instead.
+	ft fieldType
+	v  any // Component | Data | Pointer
 
-	// Pointer to the corresponding tree node. Can be nil for the just assigned fields.
+	// Pointer to the corresponding tree node. Can be nil for the just created fields.
 	node *Node
 }
 
-func (fi fieldInternal) IsEmpty() bool {
-	return fi.value == nil
+func (fi fieldInternal) isEmpty() bool {
+	return fi.value() == nil
 }
 
-func (f Field[T]) Get(Context) (T, error) {
-	// remember to handle d == nil case
+func (fi fieldInternal) value() any {
+	if fi.node == nil {
+		return fi.v
+	}
+	return fi.node.value
+}
 
-	panic("not implemented")
+func (fi fieldInternal) fieldType() fieldType {
+	if fi.node == nil {
+		return fi.ft
+	}
+	return fi.node.fieldType()
+}
+
+func (f Field[T]) Get(chasmContext Context) (T, error) {
+	// If value is not nil, use it.
+	if f.Internal.v != nil {
+		return f.Internal.v.(T), nil
```

Comment:
just to confirm. For newly created Fields, after closing the transaction, will the `f.Internal.v` field will be reset to nil? so that later Get() will be served by the node?

Reply:
Good point. Yes, I will clear it there, but I am also changing order here. `node` is now taking precedence and will be used if not `nil`.